package/files/nginx-pubcloud/nginx-registry-https.conf

```diff
@@ -0,0 +1,138 @@
+upstream rmt {
+    server 127.0.0.1:4224;
+}
+
+upstream registry {
+    server 127.0.0.1:5000;
+}
+
+## Set a variable to help us decide if we need to add the
+## 'Docker-Distribution-Api-Version' header.
+## The registry always sets this header.
+## In the case of nginx performing auth, the header is unset
+## since nginx is auth-ing before proxying.
+map $upstream_http_docker_distribution_api_version $docker_distribution_api_version {
+  '' 'registry/2.0';
+}
+
+
+server {
+    listen 443 default_server  ssl;
+    listen [::]:443 default_server ssl;
+    server_name rmt;
+
+    access_log  /var/log/nginx/rmt_https_access.log;
+    error_log   /var/log/nginx/rmt_https_error.log;
+    root        /usr/share/rmt/public;
+
+    ssl_certificate     /etc/rmt/ssl/rmt-server.crt;
+    ssl_certificate_key /etc/rmt/ssl/rmt-server.key;
+
+    ssl_protocols       TLSv1.2 TLSv1.3;
+
+    include /etc/nginx/rmt-auth*.d/auth-location*.conf;
+
+    location / {
+        autoindex off;
+    }
+
+    # no authentication for license files
+    location ~ ^.*/product.license/.*$ {
+        autoindex on;
+        log_not_found off;
+    }
+
+    location /repo {
+        autoindex on;
+        log_not_found off;
+        include /etc/nginx/rmt-auth*.d/auth-handler*.conf;
+    }
+
+    location = /repo/repoindex.xml {
+        try_files $uri @rmt_app;
+    }
+
+    location /connect {
+        try_files $uri @rmt_app;
+    }
+
+    location /services {
+        try_files $uri @rmt_app;
+    }
+
+    location /api {
+        try_files $uri @rmt_app;
+    }
+
+    location /v2 {
+        try_files $uri @rmt_app;
+    }
+
+    location @rmt_app {
+        proxy_pass          http://rmt;
+        proxy_redirect      off;
+        proxy_read_timeout  600;
+
+        proxy_set_header Host $http_host;
+        proxy_set_header X-Forwarded-Proto $scheme;
+        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
+        proxy_set_header X-Forwarded-Ssl on;
+        proxy_set_header X-Real-IP $remote_addr;
+    }
+
+    # An alias to RMT CA certificate, so that it can be downloaded to client machines.
+    location /rmt.crt {
+        alias /etc/rmt/ssl/rmt-ca.crt;
+    }
+
+    # smt.crt is used by cloud-regionsrv-client < 8.1.1
+    location /smt.crt {
+        alias /etc/rmt/ssl/rmt-ca.crt;
+    }
+}
+
+server {
+    listen 443   ssl;
+    listen [::]:443 ssl;
+    server_name ~^registry;
```

Comment:
Do we not need an auth setup for the registry path?

Reply:
No, the auth is defined in the registry config.